parsers.js

```
@@ -74,13 +74,14 @@ PARSERS.parse_name = function parse_name (str, data) {
 
     if (name[0] === '[' && name[name.length - 1] === ']') {
       res.optional = true
-      name = name.slice(1, -1)
 
-      if (name.indexOf('=') !== -1) {
-        const parts = name.split('=')
-        name = parts[0]
-        res.default = parts[1].replace(/^(["'])(.+)(\1)$/, '$2')
-      }
+      const result = name.slice(1, -1).match(
+        /^\s*([^=]+?)(?:\s*=\s*(.+?))?\s*(?=$)/
+      )
```

@jaydenseric good point, this should really be less restrictive. Would you mind if we avoid the RegExp to keep it simpler. Maybe something like
```js
name = name.slice(1, -1)
const eq = indexOf('=')
if (eq !== -1) {
  name = name.slice(0, eq).trim()
  res.default = name.slice(eq + 1).trim()
}
```
other than that I would merge it right away

---

I can sympathise with regex being a little tricky to follow, but one of the nice things about having a comprehensive regex is that you can tell up front if the name is valid syntax or not. I hadn't done it, but you could do something like:

```js
const result = name.slice(1, -1).match(/^\s*([^=]+?)(?:\s*=\s*(.+?))?\s*(?=$)/);

// Add this:
if (!result) throw new SyntaxError('Invalid syntax for a name in a JSDoc tag.');
```

For example, what if the name is `[=]`? Or `[==]`? IMO it's better to throw a syntax error than return a misleading `''` or `=` name. Without a regex you would have to do a lot of little checks throught the code.

---

I just added a few more tests to the regex proving the invalid syntax names don't match: https://regex101.com/r/owjcTx/7/tests .

---

Makes sense, let’s keep the regexp